tests/collaborative/collaborative_monkey_party.test.ts

```diff
@@ -0,0 +1,310 @@
+      } else {
+        // reduce the chunk command-by-command
+        let chunkReduced = true;
+        while (chunkReduced) {
+          chunkReduced = false;
+          const chunk = actions[chunkIndex];
+          for (let cmdIndex = 0; cmdIndex < chunk.length; cmdIndex++) {
+            const testChunkCase = [...actions];
+            testChunkCase[chunkIndex] = chunk.slice(0, cmdIndex).concat(chunk.slice(cmdIndex + 1));
+            const { fail } = rerunTest(testChunkCase);
+            if (fail) {
+              actions = testChunkCase;
+              chunkReduced = true;
+            }
+          }
+        }
+      }
+    }
+  }
+  return actions;
+}
+
+function areSynced(users: Model[]): boolean {
+  for (let i = 0; i < users.length - 1; i++) {
+    if (!deepEquals(users[i].exportData(), users[i + 1].exportData())) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function evaluationsAreSynced(users: Model[]): boolean {
+  for (let i = 0; i < users.length - 1; i++) {
+    for (const sheetId of users[i].getters.getSheetIds()) {
+      const sheetZoneXc = zoneToXc(users[i].getters.getSheetZone(sheetId));
+      const valuesUserA = getRangeValues(users[i], sheetZoneXc, sheetId);
+      const valuesUserB = getRangeValues(users[i + 1], sheetZoneXc, sheetId);
+      if (!deepEquals(valuesUserA, valuesUserB)) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+function run(network: MockTransportService, users: Model[], actions: UserAction[][]) {
+  const executed: UserAction[][] = [];
+  for (const commandGroup of actions) {
+    const concurrentlyExecuted: UserAction[] = [];
+    executed.push(concurrentlyExecuted);
+    try {
+      network.concurrent(() => {
+        for (const { command, user } of commandGroup) {
+          const result = user.dispatch(command.type, command);
+          if (result.isSuccessful) {
+            concurrentlyExecuted.push({ command, user });
+          }
+        }
+      });
+      expect(evaluationsAreSynced(users)).toBe(true);
+    } catch (e) {
+      console.error(e);
+      concurrentlyExecuted.push(...commandGroup);
+      return {
+        fail: true,
+        executedActions: executed,
+      };
+    }
+  }
+  return {
+    fail: !areSynced(users),
+    executedActions: executed,
+  };
+}
+
+function randomizeCommandsPayload(commands: CoreCommand[]) {
+  const allSheetIds = commands.map((cmd) => cmd.sheetId).filter(Boolean);
+  for (const command of commands) {
+    if (isSheetDependent(command)) {
+      command.sheetId = randomChoice(allSheetIds);
+    }
+    if (isTargetDependent(command)) {
+      command.target = Array.from(range(1, randomIntFromInterval(2, 3)), generateRandomZone);
+    } else if (isPositionDependent(command)) {
+      command.col = randomIntFromInterval(0, 10);
+      command.row = randomIntFromInterval(0, 10);
+    } else if (isRangeDependant(command)) {
+      command.ranges = command.ranges.map((range) => ({
+        ...range,
```

Comment:
should also randomize the sheetId no ? Or pick the same one as the randomized cmd.sheetId most likely for CFs